packages/apollo-link-http/README.md

```diff
@@ -69,7 +69,48 @@ client.query({
 })
 ```
 
-## Upgrading from `apollo-fetch` / `apollo-client` 
+## Errors
+The Http Link draws a distinction between client, server and GraphQL errors.
+Server errors can occur in three different scenerios: parse, network and data errors.
+This list describes the different errors and what the response Observable calls:
+
+* Client parse errors: the request body is not-serializable due to circular references for example
+* Server parse errors: the response from the server cannot be parsed((response.json())[https://developer.mozilla.org/en-US/docs/Web/API/Body/json])
+* Server network errors: the response has a status of >= 300
+* Server data errors:  the parse request does not contain `data` or `errors`
+* GraphQL errors: an objects in the `errors` array for a 200 level status
+
+Since many server implementations can return a valid GraphQL result on a server network error, the thrown `Error` object contains the parsed server result.
+A server data error also receives the parsed result.
+
+This table provides a summary of the errors, the `Observable` method called by the HTTP link, and type of the error thrown.
+
+| Error   |      Callback      |Error Type |
+|----------|:-------------:|------|
+| Client Parse |    `error`   | `ClientParseError` |
+| Server Parse |    `error`   | `ServerParseError` |
+| Server Network |    `error`   | `ServerError` |
+| Server Data |    `error`   | `ServerError` |
+| GraphQL Error |    `next`   | `` |
+
+```js
+type ServerError = Error & {
+  response: Response;
+  result: Record<string, any>;
+  statusCode: number;
+};
+
+type ServerParseError = Error & {
+  response: Response;
+  statusCode: number;
+};
+
+type ClientParseError = Error & {
+  parseError: Error;
+};
+```
+
+## Upgrading from `apollo-fetch` / `apollo-client`

```

Comment:
I think `apollo-client` in this heading is a bit misleading - maybe `createNetworkInterface`?